### PR TITLE
Fix polyvox async locking

### PR DESCRIPTION
**Pull request by sethalves**
- fix a bug in PolyVox Entity which could cause a crash when switch domains or otherwise deleting the entity.

### CODE REVIEW

**Comment by sethalves** [comment]
https://app.asana.com/0/26485730907942/103479990948364
https://app.asana.com/0/26485730907942/103433202828929
and maybe
https://app.asana.com/0/26485730907942/103307223537146

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1bb94c966bf701e425e5688d6423dd59fedc5fb7 available at: https://gist.github.com/5a0182b8da2a36f3192c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7429.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7429.dmg

**Comment by sethalves** [comment]
to test:
- put a lot of polyvox in a domain (use polyvox.js) and smoke test
- switch domains back and forth
- launch interface into the domain with polyvox and close it before the polyvoxes finish loading

File: libraries/entities-renderer/src/RenderablePolyVoxEntityItem.h [hunk 1]
```diff
@@ -155,11 +155,15 @@ class RenderablePolyVoxEntityItem : public PolyVoxEntityItem {
     void compressVolumeDataAndSendEditPacket();
     void compressVolumeDataAndSendEditPacketAsync();
     void getMesh();
+    void getMeshAsyncWithSemaphore();
     void getMeshAsync();
     void computeShapeInfoWorker();
     void computeShapeInfoWorkerAsync();
 
-    QSemaphore _threadRunning{1};
+    QSemaphore _threadRunning{1}; // use a QSemaphore rather than a QMutex so lock and unlock can be on different threads
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderablePolyVoxEntityItem.h]: is there a `std::` equivalent you could use? maybe `std::condition_variable`?

Comment (sethalves) [inline libraries/entities-renderer/src/RenderablePolyVoxEntityItem.h]: Well... maybe, but it's being used to guard a group of QtConcurrents from each other.

Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderablePolyVoxEntityItem.h]: whatever you say boss

File: libraries/entities-renderer/src/RenderablePolyVoxEntityItem.cpp [hunk 2]
```diff
@@ -73,8 +73,20 @@ RenderablePolyVoxEntityItem::RenderablePolyVoxEntityItem(const EntityItemID& ent
 }
 
 RenderablePolyVoxEntityItem::~RenderablePolyVoxEntityItem() {
+    _threadRunning.acquire();
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderablePolyVoxEntityItem.cpp]: why not `lockThread()`?

**Comment by ZappoMan** [comment]
looks reasonable to me... my comments are not blockers.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 87eb3ec2a06f98b786dbaafabdb8526f78ef1079 available at: https://gist.github.com/5a0182b8da2a36f3192c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7429.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7429.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b2803eef2ea075b298ffe30af892b662b9f25c7b available at: https://gist.github.com/5a0182b8da2a36f3192c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7429.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7429.dmg

**Comment by OmegaHeron** [comment]
Tested Windows 7

General Smoke testing :+1: 
"put a lot of polyvox in a domain (use polyvox.js) and smoke test"
Placed 52 terrain pieces - no issues seen.

"switch domains back and forth" - Crashed

launch interface into the domain with polyvox and close it before the polyvoxes finish loading - Crashed before I got a chance to close before finish load.

QA Fail here - much bugsplat sent.

**Comment by sethalves** [comment]
Omega tried again and didn't see the same crashing.  He asks that others try it.

**Comment by OmegaHeron** [comment]
Try visiting hifi://heron/16044.2,16010.1,15960.4/0,0.794313,0,0.607508

First just wait for scene to load - don't move.  Then - if no crash, close Interface and re-run, this time, try rotating your AV as it loads.

**Comment by sethalves** [comment]
I'm going to take a different approach to this.  closing this PR for now, new one soon.

**Pull request closed without merging**